### PR TITLE
Feature - add linting to build steps

### PR DESCRIPTION
**Pull request by rikhilrai**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -10,14 +10,14 @@ jobs:
...
     steps:
-    - name: Check out the repo
-      uses: actions/checkout@v2
-    - name: Login to GitHub Packages Docker Registry
-      run: echo ${{ secrets.GITHUB_TOKEN }} | docker login docker.pkg.github.com -u ${{ github.repository_owner }} --password-stdin
-    - name: Build docker image
-      run: docker build --tag $DOCKER_IMAGE:$DOCKER_TAG $DIRECTORY
-    - name: Push to GitHub Packages
-      run: docker push $DOCKER_IMAGE:$DOCKER_TAG
+      - name: Check out the repo
```
Comment (rikhilrai) [inline .github/workflows/deploy.yml:15]: this was vscode formatting. happy to revert

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -8,7 +8,11 @@ jobs:
   build-app:
     runs-on: ubuntu-latest
     steps:
-    - name: Check out the repo
-      uses: actions/checkout@v2
-    - name: Build docker image
-      run: docker build $DIRECTORY
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Install dependencies
```
Comment (rikhilrai) [inline .github/workflows/pull-request.yml:14]: not sure if we want to run npm commands here or in docker?

Comment (krishan711) [reply .github/workflows/pull-request.yml:14]: would be better in docker ideally, but can replace later

File: .github/workflows/pull-request.yml [hunk 3]
```diff
@@ -2,13 +2,19 @@ name: Build
 on:
   pull_request:
     branches: [ main ]
-env:
-  DIRECTORY: ./app
+defaults:
+  run:
+    working-directory: app
```
Comment (rikhilrai) [inline .github/workflows/pull-request.yml]: replaced `DIRECTORY: ./app` with this. what are your thoughts?
`docker build $DIRECTORY` now becomes `docker build .`

Comment (krishan711) [reply .github/workflows/pull-request.yml]: very nice

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -2,24 +2,28 @@ name: Deploy
 on:
   push:
     branches: [ main ]
+defaults:
+  run:
```
Comment (krishan711) [inline .github/workflows/deploy.yml]: i think should go into the deploy-app job cos its only used there

**Review by krishan711 — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (rikhilrai) [reply .github/workflows/deploy.yml]: nice, moved 👍

**Pull request merged**